ethereum/event/BREvent.c

```diff
@@ -268,10 +288,16 @@ eventHandlerStop (BREventHandler handler) {
         // Quit the thread by aborting the queue wait.
         eventQueueDequeueWaitAbort (handler->queue);
 
-        // Wait for the thread.
+        // Wait for the thread.  We must release the lock here because the `eventHandlerThread`
+        // could itself be waiting on the lock (only as part of a running, dispatched function).
+        // Once we give the lock, the dispatched function will complete, the `eventHandlerThread`
+        // will return to the EventQueue and see the above 'Wait Abort' cond-var signaled.  At
+        // that point, `eventHandlerThread` will finish and the following 'join' returns.
+
+        // TODO: Race on 'start/stop' here
+        pthread_mutex_unlock(&handler->lock);
```

Comment:
Few issues here:
1. Flagged by TSAN as a data race
2. Can be PTHREAD_NULL, as it is set by the other thread
3. Can result in multiple simultaneous calls to pthread_join (from man page: "The results of multiple simultaneous calls to pthread_join() specifying the same target thread are undefined.")